server/app/views/questiontypes/ApplicantQuestionRendererImpl.java

```diff
@@ -78,23 +82,53 @@ public final DivTag render(ApplicantQuestionRendererParams params) {
         validationErrors.getOrDefault(question.getContextualizedPath(), ImmutableSet.of());
     if (!questionErrors.isEmpty()) {
       // Question error text
-      questionTextDiv.with(
+      questionSecondaryTextDiv.with(
           BaseHtmlView.fieldErrors(
               messages, questionErrors, ReferenceClasses.APPLICANT_QUESTION_ERRORS));
     }
 
     if (question.isRequiredButWasSkippedInCurrentProgram()) {
       String requiredQuestionMessage = messages.at(MessageKey.VALIDATION_REQUIRED.getKeyName());
-      questionTextDiv.with(
+      questionSecondaryTextDiv.with(
           div()
               .withClasses(Styles.P_1, Styles.TEXT_RED_600)
               .withText("*" + requiredQuestionMessage));
     }
 
-    return div()
-        .withId(question.getContextualizedPath().toString())
-        .withClasses(Styles.MX_AUTO, Styles.MB_8, getReferenceClass(), getRequiredClass())
-        .with(questionTextDiv)
-        .with(renderTag(params, validationErrors));
+    if (inputFieldType == InputFieldType.COMPOSITE) {
+      // Group fields with fieldset and legend for a11y.
+      LegendTag legend =
```

Comment:
Is it possible to avoid forking of this extent due to the different tag type? It makes it a bit difficult to ensure that the majority of the implementation is the same.

I believe that J2HTML has base classes for tags that can contain child elements vs not (https://javadoc.io/static/com.j2html/j2html/1.5.0/j2html/tags/ContainerTag.html).

As I see it, we have a "questionPrimaryText" element which could be either a `<legend>` or a `<div>`. Additionally, we have a container element that could either be a `<fieldset>` or a `<div>`. Could we fork these two based on whether something was composite or not and then apply the additional styling on those?

If the restructuring doesn't feel worth it or possible, feel free to ignore.

Reply:
Yes, thank you for the suggestion! I refactored this as best I could to make it easier to keep the implementations in sync.

One issue I ran into was when trying to do ContainerTag.with() - error: "unchecked call to with(....) as a member of the raw type j2html.tags.ContainerTag". So I had to keep some of the logic duplicated in the if/else blocks. I'm not sure if there's a way to be able to share that logic.